caddytls/maintain.go

```diff
@@ -334,8 +334,15 @@ func DeleteOldStapleFiles() {
 // meaning that it is not expedient to get an
 // updated response from the OCSP server.
 func freshOCSP(resp *ocsp.Response) bool {
+	nextUpdate := resp.NextUpdate
+        // If there is an OCSP responder certificate, and it expires before the
+        // OCSP response, use its expiration date as the end of the OCSP
+        // response's validity period.
+	if resp.Certificate && resp.Certificate.NotAfter.Before(nextUpdate) {
```

Comment:
This isn't JavaScript; it'll have to be `if resp.Certificate != nil && ...` :)

Otherwise this looks pretty good!